Fc 1146 event sources (#7)
Templum/developers-community: pages/Documents/DeveloperTools/LivePersonFunctions/EventSources/conversational-commands.md

indicator: both
---

> With Conversational Commands, agents are able to invoke functions from the Agent Workspace by typing `/` into the messaging input in the connections panel. This empowers agents to use a bank of preconfigured, account-specific commands, to enrich the conversation with integrated functions that will help them interact efficiently with the consumer.

Reviewer: This sounds like this would work out of the box. I would start  something like:
```suggestion
When properly set up, agents can use Conversational Commands to invoke functions from the Agent Workspace. By typing `/` into the messaging input in the connections panel, all available commands are shown. This empowers agents to use a bank of preconfigured, account-specific commands, to enrich the conversation with integrated functions that will help them interact efficiently with the consumer.
```

<img src="img/functions/functions_conversational_commands.png" alt="Functions: Conversational Commands" style="width:50%;"/>

Along with the invocation a payload is sent containing the conversation ID and the arguments provided by the agent. This payload can then be used in the function for further processing and referencing. The result of the function has to either be a string or a number and will be shown in the Agent Workspace as a message in the transcript.

**Note** The response is not actually stored in the conversation transcript, it will only be stored temporarily in the browser and is only visible to the agent.

{: .important}

> It is required that your account has the New Agent Workspace enabled; please contact your account team in order to do this.

Reviewer: Maybe move this down under the `### Configuration`heading.

Author: good idea,

### Configuration

> Agent permissions required: **Use enhanced Agent Workspace** and **FaaS-Invocation** permissions have to be enabled.

Reviewer: ```suggestion
To be able to use Conversational Commands an Agent needs the following permissions:
* **Use enhanced Agent Workspace** 
* **FaaS-Invocation**
```
Can we maybe also add a link to how to adjust user permissions?

Author: not sure if there is that info in developers docs (can 't find it) since this is normally done by an admistratior

#### Step 1 - Create a new Function

Create a new function from **Conversational Command** as the triggering event. The name of the function will be displayed as the name of the Conversational Command in the Agent Workspace.

The `description` will also be shown in the Agent Workspace. To give the Agent a hint on how to use the function, we recommend providing the available arguments for the function in square brackets in the description.

Here is an example for the description of a Conversational Command that sends the transcript of the conversation via email:

`[email, subject:optional] - Sends the transcript of the conversation to the provided email address.`

**Note:** You can create up to 30 functions for this event.

#### Step 2 - Edit the Function

Adjust the code in the template according to your needs by modifying the function. On the right side you can see an example of the payload (in the sidebar, which you might need to open).

> As mentioned above, the function must return either a string or a number. This response will be displayed in the Agent Workspace as a message. If the function returns anything else, the Agent will only be shown a generic success message. If the functionreturns an `Error`, this will be presented to the Agent as an error message.

Reviewer: ```suggestion
The function must return either a string or a number. This response will be displayed in the Agent Workspace as a message. If the function returns anything else, the Agent will only be shown a generic success message. If the function returns an `Error`, this will be presented to the Agent as an error message.
```

indicator: both
---

> With Conversational Commands, agents are able to invoke functions from the Agent Workspace by typing `/` into the messaging input in the connections panel. This empowers agents to use a bank of preconfigured, account-specific commands, to enrich the conversation with integrated functions that will help them interact efficiently with the consumer.

Reviewer 2: @eplazaso get Martin to check this specific part. As he is the daddy of the integration

<img src="img/functions/functions_conversational_commands.png" alt="Functions: Conversational Commands" style="width:100%;"/>

Along with the invocation a payload is sent containing the conversation ID and the arguments provided by the agent. This payload can then be used in the function for further processing and referencing. The result of the function has to either be a string or a number and will be shown in the Agent Workspace as a message in the transcript.

**Note** The response is not actually stored in the conversation transcript, it will only be stored temporarily in the browser and is only visible to the agent.

{: .important}
It is required that your account has the New Agent Workspace enabled; please contact your account team in order to do this.

### Configuration

> {: .important}

Reviewer 2: No need for {: .important} or notic, if it is the only thing in the chapter.
```suggestion
```

indicator: both
---

> With **Conversational Commands**, agents are able to invoke `lambdas` from the Agent Workspace by typing `/` into the messaging input in the connections panel. This empowers agents to use a bank of preconfigured, account-specific commands, to enrich the conversation with integrated functions that will help them interact efficiently with the consumer.

Reviewer 2: ```suggestion
With Conversational Commands, agents are able to invoke `functions` from the Agent Workspace by typing `/` into the messaging input in the connections panel. This empowers agents to use a bank of preconfigured, account-specific commands, to enrich the conversation with integrated functions that will help them interact efficiently with the consumer.
```

Reviewer 2: Would be careful with the wording here. It sounds almost like when you type `/` there will be a set of commands from the get-go. Which is not the case as far as I understood

Author: I think in the image is self explained

Author: <img width="962" alt="image" src="https://user-images.githubusercontent.com/44361656/156984796-dbe66043-8c93-4ae4-a77d-1de37813f611.png">

Author: it is called "commands"
